### PR TITLE
Lu 9

### PR DESCRIPTION
**Pull request by sanketXynes**
Checkbox

### CODE REVIEW

File: src/components/TextInput/textInput.type.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+// import { AssetProps } from "@icons/types/asset.types";
+
+export interface TextInputProps
+  extends React.HtmlHTMLAttributes<HTMLInputElement> {
+  type?: "fill" | "outlined" | "outline-only" | "-M";
```
Comment (archan96) [inline src/components/TextInput/textInput.type.ts]: Why this -M?

File: src/components/TextInput/textInput.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ...props
+  },
+  ref
+) => {
+  const [visible, setVisible] = useState(false);
+
+  const InputWrapper = styled.div`
+    padding: ${spacing?.padding?.small} ${spacing?.padding?.medium};
+    border-radius: ${spacing?.borderRadius?.small};
+    background-color: ${type !== "outline-only"
```
Comment (archan96) [inline src/components/TextInput/textInput.tsx]: make this bit efficient

File: src/components/Checkbox/checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    background-color: ${color?.seagreen};
+  `;
+
+  const StyledCheckbox = styled.div<{ checked: boolean }>`
+    position: relative;
+    /* display: inline-block; */
+    width: 1.2vw;
+    height: 1.2vw;
+    border: ${strokes?.s0} solid ${color?.border1};
+    background-color: ${color?.skyblue1};
```
Comment (archan96) [inline src/components/Checkbox/checkbox.tsx:33]: This is brand100 in other places.

File: src/components/TextInput/textInput.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        <TextInputContainer
+          type={!visible ? inputType : "text"}
+          className={cx("lmTextInputContainer")}
+          placeholder={placeholder}
+          ref={ref}
+          {...props}
+        />
+        {inputType === "password" ? (
+          <LMAsset
+            //   style={iconAtEnd ? { marginLeft: "0.5vw" } : { marginRight: "0.5vw" }}
```
Comment (archan96) [inline src/components/TextInput/textInput.tsx]: Remove unused comments

File: src/components/TextInput/textInput.type.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+// import { AssetProps } from "@icons/types/asset.types";
```
Comment (archan96) [inline src/components/TextInput/textInput.type.ts]: remove unused comments

**Review by archan96 — CHANGES_REQUESTED** [review]
Please make the following changes and push @sanketXynes

File: src/components/Checkbox/checkbox.type.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+
+export interface CheckboxProps
+  extends React.HtmlHTMLAttributes<HTMLInputElement> {
+  label?: string;
+  showIndeterminate?: boolean;
+  isChecked?: boolean;
```
Comment (archan96) [inline src/components/Checkbox/checkbox.type.ts:7]: It needs a callback onUpdate

**Review by archan96 — APPROVED** [review]
Cool

**Review by archan96 — APPROVED** [review]
Cool

**Review by archan96 — APPROVED** [review]
All changes are fine.

**Review by xynes-admin — APPROVED** [review]
All changes are approved.

**Review by archan96 — APPROVED** [review]
Approved resolved conflicts

**Pull request merged**